src/AppInstallerCLIE2ETests/UninstallCommand.cs

```diff
@@ -8,23 +8,101 @@ namespace AppInstallerCLIE2ETests
 
     public class UninstallCommand : BaseCommand
     {
-        private const string UninstallTestExeInstalledFile = @"TestExeUninstalled.txt";
+        private const string CustomProductCode = "{f08fc03c-0b7e-4fca-9b3c-3a384d18a9f3}";
+        private const string UninstallTestExeUninstalledFile = "TestExeUninstalled.txt";
+        private const string UninstallTestMsiInstalledFile = "AppInstallerTestExeInstaller.exe";
+        private const string UninstallTestMsixName = "6c6338fe-41b7-46ca-8ba6-b5ad5312bb0e";
 
-        //[Test]
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            InitializeAllFeatures(false);
+            ConfigureFeature("uninstall", true);
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+            InitializeAllFeatures(false);
+        }
+
+        [Test]
         public void UninstallTestExe()
         {
-            // Example Uninstall Command Test
+            // Uninstall an Exe
+            var installDir = TestCommon.GetRandomTestDir();
+            TestCommon.RunAICLICommand("install", $"{Constants.ExeInstallerPackageId} --silent -l {installDir}");
+            var result = TestCommon.RunAICLICommand("uninstall", Constants.ExeInstallerPackageId);
+            Assert.AreEqual(Constants.ErrorCode.S_OK, result.ExitCode);
+            Assert.True(result.StdOut.Contains("Successfully uninstalled"));
+            Assert.True(VerifyTestExeUninstalled(installDir));
+        }
+
+        [Test]

```

Comment:
>[Test] [](start = 8, length = 6)

Should we comment out the tag instead of checking for ignoring since msi project is never supported in azp?

Reply:
I doubt anybody would ever run this test if the tag is commented out. If we keep it but ignore it, even if it never runs in azp it would at least run if somebody runs it locally and sets up the MSI project. The downside is that the pass rate for the tests in azp goes down from 100% with this marked as "Other".

Reply:
ok, if we decide to go with this "ignore" route, please update the InstallMSI() in InstallCommand test to also do the ignoring and uncomment the tag. Thanks.

---
In reply to: [543545150](https://github.com/microsoft/winget-cli/pull/659#discussion_r543545150) [](ancestors = 543545150)